tests/integration/solidity_contracts/conftest.py

```diff
@@ -140,20 +145,27 @@ async def _factory(contract_app, contract_name, *args, **kwargs):
 
 
 @pytest.fixture(scope="package")
-def create_account_with_bytecode(starknet, kakarot, deploy_bytecode, deploy_eoa):
+def create_account_with_bytecode_and_storage(
+    starknet, kakarot, deploy_bytecode, deploy_eoa
+):
     """
     Fixture to create a solidity contract in kakarot without running the bytecode.
-    The given bytecode is directly stored into the account, similarly to what is done
-    in a genesis config.
+    The given bytecode and storage are directly stored into the account, similarly to
+    what is done in a genesis config.
 
     Returns the corresponding starknet contract with the extra evm_contract_address attribute.
     """
 
-    async def _factory(bytecode: str, caller_eoa=None):
+    async def _factory(
+        bytecode: str = "", storage: Optional[Dict[str, str]] = None, caller_eoa=None
```

Comment:
```suggestion
        bytecode: str = "", storage: Optional[Dict[str, str]] = {}, caller_eoa=None
```

Reply:
never put mutable arg as default in python because their updates would be propagated to the next call.

```
def fun(d={}):
    d['a'] = 1 + d.get('a', 0)
    return d


fun()
# {'a': 1}
fun()
# {'a': 2}
```

Reply:
Ahhh did not know making the default {} made it mutable

Reply:
in python, everything is an object, and if you do `dir(fun)` you'll see all the attributes of the `fun` object.
More precisely, you'll have `fun.__defaults__`, and indeed

```
> fun.__defaults__
({},)
> fun()
> fun.__defaults__
({'a': 1},)
```

this is because the `__defaults__` points to the same dict, which has been muted